juju/main.py

```diff
@@ -0,0 +1,218 @@
+#!/usr/bin/env python3
+
+import os
+import sys
+import yaml
+import importlib
+
+CHARM_STATE_FILE = '.unit-state.db'
+
+
+def debugf(format, *args, **kwargs):
+    pass
+
+
+class InvalidHookEntry(Exception):
+    pass
+
+
+def _get_charm_dir():
+    charm_dir = os.environ.get("JUJU_CHARM_DIR")
+    if charm_dir is None:
+        # Assume $JUJU_CHARM_DIR/lib/juju/main.py structure.
+        charm_dir = os.path.dirname(os.path.dirname(
+            os.path.dirname(os.path.abspath(__file__))))
+    return charm_dir
+
+
+def _load_metadata():
+    with open('metadata.yaml') as f:
+        metadata = yaml.load(f, Loader=yaml.SafeLoader)
+    return metadata
+
+
+def _get_charm_endpoints():
+    """Get a list of endpoints from the charm metadata.
+    """
+    metadata = _load_metadata()
+    charm_endpoints = list()
+    for endpoint_type in ['provides', 'requires', 'peers']:
+        endpoints = metadata.get(endpoint_type)
+        if endpoints:
+            charm_endpoints.extend(endpoints)
+    return charm_endpoints
+
+
+def _handle_event_link(charm_dir, event_name):
+    event_hook_path = os.path.join(
+        charm_dir,
+        'hooks', event_name.replace('_', '-'))
```

Comment:
I personally find the [pathlib.Path()](https://docs.python.org/3/library/pathlib.html) object-oriented style syntax a bit nicer to work with, but it's just an aesthetic thing.  Just wanted to raise it as an option, but of course whatever we go with, we should be consistent.

Reply:
+1

Reply:
OK, I'll change it.